src/main/java/mat/server/service/FhirValidationReportService.java

```diff
@@ -212,11 +205,64 @@ private void addConversionStatusMessage(ConversionResultDto conversionResultDto,
         if (CollectionUtils.isNotEmpty(conversionResultDto.getValueSetConversionResults())) {
             for (ValueSetConversionResults valueSetConversionResults : conversionResultDto.getValueSetConversionResults()) {
                 if (Boolean.FALSE.equals(valueSetConversionResults.getSuccess())) {
-                    valueSetFhirValidationErrors = valueSetConversionResults.getValueSetFhirValidationResults();
+                    valueSetFhirValidationErrors.addAll(valueSetConversionResults.getValueSetFhirValidationResults());
                 }
             }
         }
         return valueSetFhirValidationErrors;
     }
 
+    private void getLibraryErrors(ConversionResultDto conversionResultDto, List<FhirValidationResult> libraryFhirValidationErrors, HashMap<String, List<Object>> qdmCqlConversionErrorsMap,
+                           HashMap<String, List<Object>> fhirCqlConversionErrorsMap, Map<String, Object> paramsMap, Map<String, List<CqlConversionError>> externalErrorsMap) {
+         // Library FHIR validation errors
+         CqlConversionResult cqlConversionResult;
+         if (CollectionUtils.isNotEmpty(conversionResultDto.getLibraryConversionResults())) {
+             for (LibraryConversionResults results : conversionResultDto.getLibraryConversionResults()) {
+                 if ("Not Found in Hapi".equals(results.getReason())) {
+                     paramsMap.put("LibraryNotFoundInHapi", results.getReason());
+                 }
+                 if (results.getSuccess() == null || !results.getSuccess()) {
+                     if (CollectionUtils.isNotEmpty(results.getLibraryFhirValidationResults())) {
+                         libraryFhirValidationErrors.addAll(results.getLibraryFhirValidationResults());
```

Comment:
Cyclomatic complexity is quite high.  Not sure about Codacity but SonarQube would show a warning..
Is it possible to simplify that method and extract additional sub-calls?